### PR TITLE
NOX-281 Add Validation for the commands where <OwnedEntity>UpsertDto is used

### PR DESCRIPTION
**Pull request by anna-naumova25**

### CODE REVIEW

File: src/Nox.Generator/Presentation/Api/OData/EntityControllerGenerator.cs [hunk 1]
```diff
@@ -399,6 +369,15 @@ private static void GenerateChildrenPatch(NoxSolution solution, EntityRelationsh
             }}           
         }}");
         code.AppendLine();
+
+        if (child.Keys.Any())
+        {
+            code.AppendLine($"if(!updateProperties.ContainsKey(\"{child.Keys[0].Name}\"))");
+            code.StartBlock();
+            code.AppendLine($"return BadRequest(\"{child.Keys[0].Name} is required.\");");
```
Comment (anna-naumova25) [inline src/Nox.Generator/Presentation/Api/OData/EntityControllerGenerator.cs]: Will need to apply latest changes for Exception here once they are merged

File: src/Nox.Generator/Presentation/Api/OData/EntityControllerGenerator.cs [hunk 2]
```diff
@@ -304,20 +304,9 @@ private static void GenerateChildrenPut(NoxSolution solution, EntityRelationship
...
-        }
-        else
-        {
-            code.AppendLine($"[HttpPut(\"{solution.Infrastructure.Endpoints.ApiRoutePrefix}/{parent.PluralName}/{PrimaryKeysAttribute(parent)}/{relationshipName}/{PrimaryKeysAttribute(child, "relatedKey")}\")]");
-            code.AppendLine($"public virtual async Task<ActionResult<{child.Name}Dto>> PutTo{child.PluralName}NonConventional(" +
-                $"{GetPrimaryKeysRoute(parent, solution, attributePrefix: "")}, " +
-                $"{GetPrimaryKeysRoute(child, solution, "relatedKey", "")}, " +
-                $"[FromBody] {child.Name}UpsertDto {child.Name.ToLowerFirstChar()})");
-        }
+        code.AppendLine($"public virtual async Task<ActionResult<{child.Name}Dto>> PutTo{relationshipName}(" +
```
Comment (rochar) [inline src/Nox.Generator/Presentation/Api/OData/EntityControllerGenerator.cs:307]: we should use the related entity name? to make it consistent with "normal" Relations
for owned we do not have the case of relating to the same entity

And we should keep the plural if ToMany and singular if to one?

We have other issue to rename the OwnedEntitites Navigation properties, and think we should align also the endpoints

Comment (rochar) [inline src/Nox.Generator/Presentation/Api/OData/EntityControllerGenerator.cs]: Another question, do you think its easy to migrate this to the template for owned, now that you are changing this?

File: src/Nox.Generator/Application/Factories/EntityFactory.template.cs [hunk 3]
```diff
@@ -71,7 +74,7 @@ public virtual void PartialUpdateEntity({{entity.Name}}Entity entity, Dictionary
             {{- if !IsNoxTypeCreatable key.Type || key.Type == "Guid" -}}
                 {{ continue; -}}
             {{- end }}
-        entity.{{key.Name}} = {{entity.Name}}Metadata.Create{{key.Name}}(createDto.{{key.Name}}{{if entity.IsOwnedEntity}}!{{end}});
+        entity.{{key.Name}} = {{entity.Name}}Metadata.Create{{key.Name}}(createDto.{{key.Name}}{{if entity.IsOwnedEntity}}.NonNullValue<{{keyType key}}>(){{end}});
```
Comment (rochar) [inline src/Nox.Generator/Application/Factories/EntityFactory.template.cs:77]: now looking to this, we can not have an owned used as a Foreign key, if this is possible in Nox.Solution that is incorrect.

@anna-naumova25  lets discuss this use case

**Review by rochar — CHANGES_REQUESTED** [review]
looks good, lets just discuss some points

File: src/Nox.Generator/Presentation/Api/OData/EntityControllerGenerator.cs [hunk 4]
```diff
@@ -304,20 +304,9 @@ private static void GenerateChildrenPut(NoxSolution solution, EntityRelationship
         var isSingleRelationship = relationship.WithSingleEntity();
         var relationshipName = parent.GetNavigationPropertyName(relationship);
```
Comment (rochar) [inline src/Nox.Generator/Presentation/Api/OData/EntityControllerGenerator.cs]: rename to navigationPropertyName

**Review by rochar — CHANGES_REQUESTED** [review]

File: src/Nox.Generator/Presentation/Api/OData/EntityControllerGenerator.cs (hunk 1 above)
Comment (anna-naumova25) [reply src/Nox.Generator/Presentation/Api/OData/EntityControllerGenerator.cs]: done

File: src/Nox.Generator/Presentation/Api/OData/EntityControllerGenerator.cs (hunk 4 above)
Comment (anna-naumova25) [reply src/Nox.Generator/Presentation/Api/OData/EntityControllerGenerator.cs]: done

**Review by rochar — APPROVED** [review]

**Pull request merged**